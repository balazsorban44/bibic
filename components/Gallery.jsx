import * as React from 'react'

export const Gallery = ({images}) => {
  return (
    <div className="gallery">
      {images.map((itemProps, key) =>
        <GalleryItem
          key={key}
          {...itemProps}
        />
      )
      }
    </div>
  )
}


export default Gallery


export const GalleryItem = ({
  SIZE_1024, SIZE_1440, SIZE_640, desc, title
}) =>
  desc ?
    <div className="gallery-item">
      <div className="img-wrapper">
        <GalleryImage
          alt={desc}
          sizes={{
            SIZE_640,
            SIZE_1024,
            SIZE_1440
          }}
        />
      </div>
      <h3>{title}</h3>
      <p>{desc}</p>
    </div> :
    <GalleryImage
      alt={title}
      sizes={{
        SIZE_640,
        SIZE_1024,
        SIZE_1440
      }}
    />


export const GalleryImage = ({sizes: {
  SIZE_640, SIZE_1024, SIZE_1440
}, alt}) =>
  <picture>
    <source
      media="(min-width: 540px)"
      srcSet={SIZE_1024}
    />
    <source
      media="(min-width: 1280px)"
      srcSet={SIZE_1440}
    />
    <img
      alt={alt}
      src={SIZE_640}
    />
  </picture>