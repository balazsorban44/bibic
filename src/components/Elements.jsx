import React, {useState, useEffect} from 'react'
import {Link} from 'react-router-dom'
// For the slider elements
import {colors} from '../utils/colors'
import {useTranslation} from 'react-i18next'

export const Loading = ({fullPage, timeout = 10000}) => {

  const [timedOut, setTimedOut] = useState()

  useEffect(() => {
    const tick = () => setTimedOut(true)
    const timeoutId = setTimeout(tick, timeout)
    return () => clearTimeout(timeoutId)
  }, [timeout])

  const [t] = useTranslation()
  if (timedOut) {
    return (
      <p className="not-available">
        {t("content-unavailable")}
      </p>
    )
  }

  const style = fullPage ? {
    width: "100vw",
    height: "100vh",
    display: "grid",
    justifyItems: "center",
    alignItems: "center",
    backgroundColor: colors.light
  } : undefined

  return (
    <div style={style}>
      <div className="loading"><div className="spinner"/></div>
    </div>
  )


}

export const Button = ({label, to}) => <Link {...{to}} >{label}</Link>