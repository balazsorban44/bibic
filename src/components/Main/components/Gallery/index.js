import React, {Component} from 'react'
import Slider from 'react-slick'

export default class Gallery extends Component {

  render() {
    return (
      <section id="galeria">
      <h2>Gal√©ria</h2>
      <Slider
        lazyLoad
        speed={800}
        className="gallery-slider"
      >
        {Array(30).fill(null).map((e, i) => (
          <img key={i} className="gallery-slide" alt="" src={`https://bibic-vendeghazak.github.io/web/assets/images/gallery/${i}.jpg`}/>
        ))}
      </Slider>
    </section>
    )
  }
}