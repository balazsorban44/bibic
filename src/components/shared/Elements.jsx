import React from 'react'

// For the slider elements
import arrow from '../../assets/icons/arrow.svg'
import swipe from '../../assets/icons/swipe.svg'

export const Prev = ({onClick}) => (
  <div
    className="slider-arrow slider-arrow-prev"
    onClick={onClick}
  ><img alt="" src={arrow}/></div>
)

export const Next = ({onClick}) => (
  <div
    className="slider-arrow slider-arrow-next"
    onClick={onClick}
  ><img alt="" src={arrow}/></div>
)

export const SwipeIcon = (props, {isShowingSwipe = true}) => <img {...props} className={`swipe-icon ${!isShowingSwipe ? "is-touched": ""}`} alt="" src={swipe}/>




export const Loading = () => <div className="loading"><div className="spinner"/></div>