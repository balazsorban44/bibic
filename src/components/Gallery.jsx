import React from 'react'
import {Loading} from './Elements'
import {useGallery} from 'context/gallery'
import {useTranslation} from 'react-i18next'

export const Gallery = ({count, section}) => {

  const {getGallery, loading} = useGallery()

  if (loading) {
    return (
      <div className="gallery">
        <Loading/>
      </div>
    )
  }

  const gallery = getGallery(section, count)

  return (
    <div className="gallery">
      { gallery.map(item => <GalleryItem {...item} /> ) }
    </div>
  )
}


export default Gallery


export const GalleryItem = ({SIZE_1024, SIZE_1440, SIZE_640, desc, title}) => {
  const sizes = {
    SIZE_640,
    SIZE_1024,
    SIZE_1440
  }

  const {i18n: {language}} = useTranslation()
  const alt = title?.[language] ?? desc?.[language]

  if (desc) {
    return (
      <div className="gallery-item">
        <div className="img-wrapper">
          <GalleryImage
            alt={alt}
            sizes={sizes}
          />
        </div>
        <h3>{title[language]}</h3>
        <p>{desc[language]}</p>
      </div>
    )
  }

  return (
    <GalleryImage alt={alt} sizes={sizes} />
  )
}


export const GalleryImage = ({sizes, alt}) => {
  return (
    <picture>
      <source media="(min-width: 540px)" srcSet={sizes.SIZE_1024}/>
      <source media="(min-width: 1280px)" srcSet={sizes.SIZE_1440}/>
      <img alt={alt} src={sizes.SIZE_640} />
    </picture>
  )
}