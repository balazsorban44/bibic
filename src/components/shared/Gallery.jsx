import React from 'react'
import {Loading} from './Elements'
import {withStore} from '../db'

export const Gallery = ({
  galleries, count, path
}) => {
  path = path.replace("/", "")

  const children = galleries[path] ?
    Object
      .values(galleries[path])
      .slice(0, count)
      .map((itemProps, key) =>
        <GalleryItem
          key={key}
          {...itemProps}
        />
      ) :
    <Loading/>

  return (
    <div className="gallery">{children}</div>
  )
}


export default withStore(Gallery)


export const GalleryItem = ({
  SIZE_1024, SIZE_1440, SIZE_640, desc, title
}) =>
  desc ?
    <div className="gallery-item">
      <div className="img-wrapper">
        <GalleryImage
          alt={desc}
          sizes={{
            SIZE_640,
            SIZE_1024,
            SIZE_1440
          }}
        />
      </div>
      <p>{desc}</p>
    </div> :
    <GalleryImage
      alt={title}
      sizes={{
        SIZE_640,
        SIZE_1024,
        SIZE_1440
      }}
    />


export const GalleryImage = ({sizes: {
  SIZE_640, SIZE_1024, SIZE_1440
}, alt}) =>
  <picture>
    <source
      media="(min-width: 960px)"
      srcSet={SIZE_1024}
    />
    <source
      media="(min-width: 1280px)"
      srcSet={SIZE_1440}
    />
    <img
      alt={alt}
      src={SIZE_640}
    />
  </picture>