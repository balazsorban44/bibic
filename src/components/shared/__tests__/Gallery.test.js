import {Gallery, GalleryItem} from "../Gallery"
import {Loading} from "../Elements"


describe("GalleryItem component", () => {
  const props = {
    title: "title",
    desc: "desc",
    SIZE_640: "SIZE_640.jpg",
    SIZE_1024: "SIZE_1024.jpg",
    SIZE_1440: "SIZE_1440.jpg"
  }
  const wrapper = shallow(<GalleryItem {...props}/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe("source images are defined", () => {
    it("<960px", () => {
      expect(wrapper.findWhere(e => e.prop("srcSet") === props.SIZE_1024).length).toBe(1)
    })
    it("<1280px", () => {
      expect(wrapper.findWhere(e => e.prop("srcSet") === props.SIZE_1440).length).toBe(1)
    })
    describe("default image", () => {
      it("has src", () => {
        expect(wrapper.find("img").prop("src")).toBe(props.SIZE_640)
      })

      it("alt is desc", () => {
        expect(wrapper.find("img").prop("alt")).toBe(props.desc)
      })
      it("default alt is title", () => {
        wrapper.setProps({desc: ""})
        expect(wrapper.find("img").prop("alt")).toBe(props.title)
      })
    })
  })

})

describe("Gallery component", () => {
  const props = {galleries: {}, path: "/path"}
  const wrapper = shallow(<Gallery {...props}/>)

  it("renders correctly", () => {
    expect(wrapper).toHaveLength(1)
  })

  it("shows Loading when no items", () => {
    expect(wrapper.find(Loading).length).toBe(1)
  })

  it("renders elements", () => {
    wrapper.setProps({galleries: {path: [0,1]}})
    expect(wrapper.prop("children").length).toBe(2)
  })

})

