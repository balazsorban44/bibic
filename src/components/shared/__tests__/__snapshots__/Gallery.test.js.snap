// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GalleryItem
    SIZE_1024="SIZE_1024.jpg"
    SIZE_1440="SIZE_1440.jpg"
    SIZE_640="SIZE_640.jpg"
    desc="desc"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="img-wrapper"
        >
          <GalleryImage
            alt="desc"
            sizes={
              Object {
                "SIZE_1024": "SIZE_1024.jpg",
                "SIZE_1440": "SIZE_1440.jpg",
                "SIZE_640": "SIZE_640.jpg",
              }
            }
          />
        </div>,
        <p>
          desc
        </p>,
      ],
      "className": "gallery-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <GalleryImage
            alt="desc"
            sizes={
              Object {
                "SIZE_1024": "SIZE_1024.jpg",
                "SIZE_1440": "SIZE_1440.jpg",
                "SIZE_640": "SIZE_640.jpg",
              }
            }
          />,
          "className": "img-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "desc",
            "sizes": Object {
              "SIZE_1024": "SIZE_1024.jpg",
              "SIZE_1440": "SIZE_1440.jpg",
              "SIZE_640": "SIZE_640.jpg",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "desc",
        },
        "ref": null,
        "rendered": "desc",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="img-wrapper"
          >
            <GalleryImage
              alt="desc"
              sizes={
                Object {
                  "SIZE_1024": "SIZE_1024.jpg",
                  "SIZE_1440": "SIZE_1440.jpg",
                  "SIZE_640": "SIZE_640.jpg",
                }
              }
            />
          </div>,
          <p>
            desc
          </p>,
        ],
        "className": "gallery-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <GalleryImage
              alt="desc"
              sizes={
                Object {
                  "SIZE_1024": "SIZE_1024.jpg",
                  "SIZE_1440": "SIZE_1440.jpg",
                  "SIZE_640": "SIZE_640.jpg",
                }
              }
            />,
            "className": "img-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "desc",
              "sizes": Object {
                "SIZE_1024": "SIZE_1024.jpg",
                "SIZE_1440": "SIZE_1440.jpg",
                "SIZE_640": "SIZE_640.jpg",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "desc",
          },
          "ref": null,
          "rendered": "desc",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
